JustSaying/Fluent/ISubscriptionBuilder`1.cs

```diff
@@ -0,0 +1,20 @@
+using JustSaying.Models;
+
+namespace JustSaying.Fluent
+{
+    /// <summary>
+    /// Defines a builder for a subscription.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of the messages to subscribe to.
+    /// </typeparam>
+    internal interface ISubscriptionBuilder<out T>
```

Comment:
Looks like this type argument is superfluous?

Reply:
It's not - if it isn't there, [this code](https://github.com/justeat/JustSaying/blob/624a11e33b462c830c18c5915fc7ca6d1393643a/JustSaying/Fluent/SubscriptionsBuilder.cs#L101) doesn't work.

Reply:
But you could remove it from that code too, not so? In fact, that looks like it won't even work properly anyway, as it ends up calling `.WithMessageHandler<T>(resolver)` where `T` is `Message` rather than the specific message type.

Reply:
No, because I need the type to register the handlers. The types are the specific type (see the test). The internals use `out T` so it doesn't matter what the concrete type is, it's just calling a method on a type to do building.